practice_assessments/file_storage/test_simulation.py

```diff
@@ -51,7 +51,7 @@ def test_group_3(self):
   
     def test_group_4(self):
         output = simulate_coding_framework(self.test_data_4)
-        self.assertEqual(output, ["uploaded at Initial.txt", "uploaded at Update1.txt", "got at Initial.txt", "copied at Update1.txt to Update1Copy.txt", "uploaded at Update2.txt", "rollback to 2021-07-01T12:10:00", "got at Update1.txt", "got at Initial.txt", "found at [Update1.txt, Update1Copy.txt, Update2.txt]", "got at Update2.txt"])
+        self.assertEqual(output, ["uploaded at Initial.txt", "uploaded at Update1.txt", "got at Initial.txt", "copied at Update1.txt to Update1Copy.txt", "uploaded at Update2.txt", "rollback to 2021-07-01T12:10:00", "got at Update1.txt", "got at Initial.txt", "found at [Update1.txt, Update1Copy.txt]", "file not found"])
```

Comment:
This is still incorrect. Update1Copy.txt should not be in the result for the search operation either, because it was copied at 2021-07-01T12:15:00, and then deleted in the rollback to 2021-07-01T12:10:00.

Reply:
```suggestion
        self.assertEqual(output, ["uploaded at Initial.txt", "uploaded at Update1.txt", "got at Initial.txt", "copied at Update1.txt to Update1Copy.txt", "uploaded at Update2.txt", "rollback to 2021-07-01T12:10:00", "got at Update1.txt", "got at Initial.txt", "found at [Update1.txt]", "file not found"])
```

Indeed, `Update1Copy.txt` should also not appear in the search result, since it was uploaded at `12:15`, and the rollback was to `12:10`.